### DISCUSSION TITLE
How can I draw a horizontal line / rule?

### DISCUSSION THREAD

**Discussion by ad-si** [typst/typst#3708] [Q&A] [answered]
Like Markdown's `---` and HTML's `<hr>`

**Comment by frozolotl** [comment, marked as answer]
You can use `line` for that:
```typ
#line(length: 100%)
```

**Reply by ad-si** [reply]
Feels like not quite the same:
In Md and HTML it's a semantic separator, and not necessarily a line (It's often rendered as half width line, e.g. when rendering it with Pandoc. Could also be rendered as `  *  *  *  `.

But if there's nothing else, I take it. 😛
Thanks!

**Reply by frozolotl** [reply]
You can, of course, define a function:
```typ
#let hrule = line(length: 100%)

Lorem
#hrule
Ipsum
```

If you dislike the style, you can replace it, without having to change the content:
```typ
#let hrule = align(center, line(length: 60%))
```
```typ
#let hrule = align(center, box(width: 50%, repeat[\* #h(1em)]))
```

**Reply by inferiorhumanorgans** [reply]
Is there any way to control the spacing between a horizontal rule and the text below it?

**Reply by zhsaile** [reply]
> Is there any way to control the spacing between a horizontal rule and the text below it?

#v(1em)

**Comment by m0tay** [comment]
Based on others responses I've come up with this one and I'm quite satisfied, you might like it too:

```typst
#let hrule = align(center, for i in range(3) [\* #h(1em)])
```
